bokeh/io.py

```diff
@@ -57,10 +57,29 @@
 #-----------------------------------------------------------------------------
 
 class _CommsHandle(object):
+
+    _json = {}
+
     def __init__(self, comms, doc, json):
+        self._cellno = None
+        try:
+            from IPython import get_ipython
+            ip = get_ipython()
+            hm = ip.history_manager
+            p_prompt = list(hm.get_tail(1, include_latest=True))[0][1]
```

Comment:
What versions of IPython/Jupyter API this covers? We may want to start being explicit about what we support. Otherwise, the later `except` statement (with or without `Exception`) will make it hard to understand what's going on (a bug, an unsupported version of API, ...).